src/desktop/lib/openAuthModal.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  ModalType,
+  ModalOptions,
+} from "reaction/Components/Authentication/Types"
+import { data as sd } from "sharify"
+
+const mediator = require("desktop/lib/mediator.coffee")
+
+/**
+ * Open authentication modal via 'click' trigger
+ */
+export const openAuthModal = (mode: ModalType, options: ModalOptions) => {
+  mediator.trigger("open:auth", {
+    mode,
+    ...options,
+  })
+}
```

Comment:
From 7-17 is exactly the intermediate pattern that I'm imaging for Reaction; once things are funneled into a line 7 kind of location we can remove props passing from everywhere.